prow/cmd/hmac/fakeghhook/fakeghhook.go

```diff
@@ -123,6 +126,52 @@ func (f *FakeClient) DeleteRepoHook(org, repo string, id int, req github.HookReq
 	return deleteHook(f.RepoHooks, org+"/"+repo, req)
 }
 
+func (f *FakeClient) ListCurrentUserRepoInvitations() ([]github.UserRepoInvitation, error) {
+	return f.UserRepoInvitations, f.getNextError()
+}
+
+func (f *FakeClient) AcceptUserRepoInvitation(invitationID int) error {
+	err := f.getNextError()
+	if err != nil {
+		return err
+	}
+	// Must exist
+	found := -1
+	for idx, iv := range f.UserRepoInvitations {
+		if iv.InvitationID == invitationID {
+			found = idx
+			break
+		}
+	}
+	if found != -1 {
+		f.UserRepoInvitations = append(f.UserRepoInvitations[:found], f.UserRepoInvitations[found+1:]...)
+	}
```

Comment:
Hmm, I don't really see how this `getNextError` pattern is better then. Doesn't that require us to anticipate every error in the test cases rather than catching errors we didn't anticipate? For example, if we try to accept an invite that does not exist in a unit test and we didn't instrument the test case to emit an error because we didn't expect it, we would incorrectly pass the test case. Basically it seems like making the error generation entirely manual causes us to skip validation everywhere we don't expect to need it.

Reply:
`getNextError` can better control what error is returned for these github API wrappers, for example it can return error types other than 404 not found. But I agree that it's an overkill here, will remove `getnextError` here